erpnext/accounts/doctype/journal_entry/journal_entry.py

```diff
@@ -363,11 +363,11 @@ def validate_total_debit_and_credit(self):
 	def set_total_debit_credit(self):
 		self.total_debit, self.total_credit, self.difference = 0, 0, 0
 		for d in self.get("accounts"):
-			if d.debit and d.credit:
+			if d.debit_in_account_currency and d.credit_in_account_currency:
 				frappe.throw(_("You cannot credit and debit same account at the same time"))
 
-			self.total_debit = flt(self.total_debit) + flt(d.debit, d.precision("debit"))
-			self.total_credit = flt(self.total_credit) + flt(d.credit, d.precision("credit"))
+			self.total_debit = flt(self.total_debit) + flt(d.debit_in_account_currency, d.precision("debit"))
```

Comment:
This is not correct. In the case of a multi-currency journal entry, if debit and credit in different currencies, total debit and credit will not match and lead to difference.